engines/docker/config.go

```diff
@@ -0,0 +1,26 @@
+package dockerengine
+
+import (
+	schematypes "github.com/taskcluster/go-schematypes"
+)
+
+type configType struct {
+	DockerSocket   string `json:"dockerSocket"`
+	MaxConcurrency int    `json:"maxConcurrency"`
+}
+
+var configSchema = schematypes.Object{
+	Properties: schematypes.Properties{
+		"dockerSocket": schematypes.String{
+			Title: "Docker Endpoint",
+			Description: "dockerEndpoint is the endpoint to use for communicating\n" +
+				"with the Docker daemon.",
+		},
+		"maxConcurrency": schematypes.Integer{
+			Title: "Max Concurrency",
+			Description: "maxConcurrency defines the maximum number of tasks \n" +
+				"that may run concurrently on the worker.",
+			Minimum: 1,
+		},
```

Comment:
@ckousik, regarding deployment... let's largely try to leave `// NOTE: ` comments with notes on potential problems.
For qemu-engine it was really neat to deploy tc-worker with docker, because using qemu-engine requires all these external dependencies like qemu, iptables, qemu-img, etc. for network configuration and setup.

For docker-engine we so far don't need any external binaries other than docker. So maybe we can say that deploying docker-engine is:
 * install in a `config.yml` file
 * install `taskcluster-worker`
 * add script to run `taskcluster-worker` as a service (or whatever)

If we only have to install `taskcluster-worker` as a single binary, and we don't have a long list of external dependencies... then maybe it's easier to not use docker...

If we need a lot of external things, we needed a few in docker-worker: https://github.com/taskcluster/docker-worker/tree/master/bin-utils
Then maybe we reconsider... it's also possible that we can embed such resources as binary blobs into the `taskcluster-worker` binary, or some other hack :) Maybe we package them as a tarball we ship along with taskcluster-worker and in config for the engine we specify where to find it (who knows).

It's also possible that we support engine config for specifying where the system root folder is mounted in the docker container that runs taskcluster-worker. That could totally be an engine config option.

-----
But the more I think about it, we might be better off not running taskcluster-worker from inside docker. As we minimize the risk that we remove the container we're running inside, or the image, or something stupid :) hehe

Finally, for volumes and proxies having access to the root system might be a lot easier than some complicated mapping of ports and file system folders.